tests/js/navButtonsTests.js

```diff
@@ -0,0 +1,85 @@
+        jqUnit[backVisible? "assertTrue" : "assertFalse"](msg + " - The back button is hidden", backButton.is(":visible"));
+        jqUnit[nextDisabled? "assertTrue" : "assertFalse"](msg + " - The next button is enabled", nextButton.is(":disabled"));
+        jqUnit[nextVisible? "assertTrue" : "assertFalse"](msg + " - The next button is shown", nextButton.is(":visible"));
+    };
+
+    gpii.tests.verifyButtons = function (that, currentPanelNum) {
+        jqUnit.assertEquals("The model value has been updated", currentPanelNum, that.model.currentPanelNum);
+
+        var msg,
+            start = that.options.panelStartNum,
+            end = that.options.panelTotalNum,
+            backButton = that.locate("back"),
+            nextButton = that.locate("next"),
+            backTooltip = that.backTooltip,
+            nextTooltip = that.nextTooltip;
+
+        if (currentPanelNum === start) {
+            msg = "On the start panel";
+            gpii.tests.verifyStates(msg, backButton, nextButton, true, false, false, true);
+            jqUnit.assertEquals(msg + " - The text on the next button is properly set", that.options.strings.start, nextButton.html());
+            jqUnit.assertEquals(msg + " - The tooltip content for the next button is properly set", that.options.strings.start, nextTooltip.model.content);
+        }
+
+        if (currentPanelNum > start && currentPanelNum < end) {
+            msg = "On a panel in btw the start and the last panels";
+            gpii.tests.verifyStates(msg, backButton, nextButton, false, true, false, true);
+            jqUnit.assertEquals(msg + " - The text on the back button is properly set", that.options.strings.back, backButton.html());
+            jqUnit.assertEquals(msg + " - The tooltip content for the back button is properly set", that.options.strings.back, backTooltip.model.content);
+            jqUnit.assertEquals(msg + " - The text on the next button is properly set", that.options.strings.next, nextButton.html());
+            jqUnit.assertEquals(msg + " - The tooltip content for the next button is properly set", that.options.strings.next, nextTooltip.model.content);
+        }
+
+        if (currentPanelNum === end) {
+            msg = "On the last panel";
+            gpii.tests.verifyStates(msg, backButton, nextButton, false, true, false, true);
+            jqUnit.assertEquals(msg + " - The text on the back button is properly set", that.options.strings.back, backButton.html());
+            jqUnit.assertEquals(msg + " - The tooltip content for the back button is properly set", that.options.strings.back, backTooltip.model.content);
+            jqUnit.assertEquals(msg + " - The text on the next button is properly set", that.options.strings.finish, nextButton.html());
+            jqUnit.assertEquals(msg + " - The tooltip content for the next button is properly set", that.options.strings.finish, nextTooltip.model.content);
+        }
```

Comment:
these all look very similar, can they be factored into a  new verification method?